core/src/main/python/wlsdeploy/tool/deploy/deployer_utils.py

```diff
@@ -177,6 +177,49 @@ def merge_lists(existing_list, model_list, separator=',', return_as_string=True)
     return result
 
 
+def is_delete_name(name):
+    """
+    Determines if the specified name is flagged for deletion with the "!" prefix.
+    :param name: the name to be checked
+    :return: True if the name is prefixed, false otherwise
+    """
+    return name.startswith("!")
+
+
+def get_delete_item_name(name):
+    """
+    Returns the WLST name of the item to be deleted.
+    Removes the "!" prefix from the name. Assumes the name has been checked with is_delete_name().
+    :param name: the prefixed model name of the item to be deleted
+    :return: the model name of the item to be deleted
+    """
+    if is_delete_name(name):
+        return name[1:]
+    return name
```

Comment:
Seems like this should return null or throw an exception.  Returning the name does not feel right.

Reply:
Fixed to throw exception